pkg/os/darwin/release_info.go

```diff
@@ -20,8 +21,9 @@ var (
 
 func AtLeast(targetVersion string) (bool, error) {
 	once.Do(func() {
-		macCurrentVersion, errGettingVersion = syscall.Sysctl("kern.osproductversion")
-		logging.Debugf("kern.osproductversion is: %s", macCurrentVersion)
+		macCurrentVersion, _, errGettingVersion = os.RunWithDefaultLocale("sw_vers", "--productVersion")
```

Comment:
typo in the flag, this should be `sw_vers -productVersion` with single `-`

running on macOS 12 i get the following o/p
```
$ sw_vers --productVersion
Usage: sw_vers [-productName|-productVersion|-buildVersion]

$ sw_vers -productVersion
12.7.4
```

Reply:
As per man page of `sw_vers` , it contain `--` instead `-` . But looks like even with `-` it provide same info :( 

```
SW_VERS(1)                                                             General Commands Manual                                                            SW_VERS(1)

NAME
     sw_vers – print macOS system version information

SYNOPSIS
     sw_vers
     sw_vers --productName
     sw_vers --productVersion
     sw_vers --productVersionExtra
     sw_vers --buildVersion
```